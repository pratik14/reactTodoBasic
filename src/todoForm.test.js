import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoForm from './todoForm';


it('renders correctly', () => {
  const tree = renderer
    .create(<TodoForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call onSubmitAction on submit', () => {
  const onSubmitSpy = jest.fn();

  const wrapper = mount(<TodoForm onSubmitAction={onSubmitSpy} />);
  wrapper.find('input').simulate('change', { target: { name: 'name', value: 'Abcd' } })
  wrapper.find('#submitButton').simulate('submit');

  setTimeout(() => {
    expect(onSubmitSpy).toHaveBeenCalled();
  }, 1000);
})
