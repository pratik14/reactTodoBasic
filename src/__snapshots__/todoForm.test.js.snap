// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      id="submitButton"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
